tests/e2e/environment-variables/chainsaw-test.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: environment-variables
+spec:
+  description: |
+    This e2e test verifies environment variable detection and handling for different languages.
+    It tests that environment variables are properly detected from container runtime vs manifest,
+    and that instrumentation preserves and modifies them correctly.
+  skipDelete: true
+  steps:
+    - name: Prepare destination
+      try:
+        - apply:
+            file: ../../common/apply/simple-trace-db-deployment.yaml
+
+    - name: Install Test Apps
+      try:
+        - apply:
+            file: install-test-apps.yaml
```

Comment:
we need just a few apps here, all the "version" ones can add complexity and noise IMO